meta-xt-control-domain-virtio/recipes-extended/launch-domain/files/launch-domain

```diff
@@ -0,0 +1,114 @@
+#!/bin/sh
+
+GUEST_DOMAIN_NAME=$1
+GUEST_DOMAIN_XEN_CONFIGURATION=$2
+GUEST_DOMAIN_ID=""
+
+create_domain() {
+    echo "Creating ${GUEST_DOMAIN_NAME}"
+    /usr/sbin/xl create ${GUEST_DOMAIN_XEN_CONFIGURATION}
+    echo "Created ${GUEST_DOMAIN_NAME}"
+
+    # Fetch Xen guest domain identifier
+    while [ "$GUEST_DOMAIN_ID" == "" ]; do
+        GUEST_DOMAIN_ID=$(xl domid ${GUEST_DOMAIN_NAME} 2>/dev/null)
+
+        if [ "$GUEST_DOMAIN_ID" = "" ]; then
+            sleep 1
+        else
+            echo "Parsed ${GUEST_DOMAIN_NAME} id is '${GUEST_DOMAIN_ID}'"
+        fi
+    done
+}
+
+clean_up_domain() {
+    if [ "$GUEST_DOMAIN_ID" != "" ]; then
+        echo "Cleaning up ${GUEST_DOMAIN_NAME} domain with id ${GUEST_DOMAIN_ID}"
+
+        # Here we re-fetch domain id by intention.
+        # If there is no domain id available, probably the user of the system
+        # has called "xl destroy" explicitly, so there is no need to call
+        # it the second time.
+        local GUEST_DOMAIN_ID_LOCAL=$(xl domid ${GUEST_DOMAIN_NAME} 2>/dev/null)
+
+        local XL_DESTROY_PID=""
+
+        if [ "$GUEST_DOMAIN_ID_LOCAL" != "" ]; then
+            echo "Destroying ${GUEST_DOMAIN_NAME}"
+            /usr/sbin/xl destroy ${GUEST_DOMAIN_NAME} &
+            XL_DESTROY_PID=$!
+            echo "Destroyed ${GUEST_DOMAIN_NAME}"
+        else
+            echo "Was not able to fetch ${GUEST_DOMAIN_NAME} domain id. Skip destroying the domain."
+        fi
+
+        echo "Updating virtio statuses in the xen-store."
+
+        for i in `xenstore-ls /libxl/${GUEST_DOMAIN_ID}/device/virtio | grep 'backend =' | awk '{gsub(/\"/, "", $3); print $3}'`
+        do
+            i=${i%\"}
```

Comment:
As you already has `awk '{gsub(/\"/, "", $3); ` you probably don't need this and the next line

Reply:
Hi @lorc, done.